module/gui/Collector.py

```diff
@@ -33,112 +37,575 @@ class CollectorModel(QAbstractItemModel):
+        """
+            called from main
+        """
+        func_start_time = self.time_msec()
+        QMutexLocker(self.mutex)
+        selection = []          # (pid, name, folder, password)
+        smodel = self.view.selectionModel()
+        for index in smodel.selectedRows(0):
+            item = index.internalPointer()
+            if isinstance(item, Package):
+                selection.append((item.id, item.data["name"], item.data["folder"], item.data["password"]))
+            elif not isinstance(item, Link):
+                raise TypeError("%s: Unknown item instance" % self.cname)
+        self.log.debug8("%s.getSelectedPackagesForEdit took %dms" % (self.cname, self.time_msec() - func_start_time))
+        return selection
+    
+    def saveViewItemStates(self):
+        func_start_time = self.time_msec()
+        # expanded/collapsed
+        for p, package in enumerate(self._data):
+            pid = package.id
+            index = self.index(p, 0)
+            if index.isValid():
+                e = self.view.isExpanded(index)
+                self.expandedState[str(pid)] = e
+                self.log.debug0("%s.saveViewItemStates:expanded:   row:%d   name:'%s'   pid:%d   e:%d" % (self.cname, p, package.data["name"], pid, e))
+            else:
+                raise ValueError("%s: Invalid index" % self.cname)
+        # current item
+        self.currentItem = None
+        smodel = self.view.selectionModel()
+        ci = smodel.currentIndex()
+        if ci.isValid():
+            item = ci.internalPointer()
+            if isinstance(item, Package):
+                pid = item.id
+                fid = None
+            elif isinstance(item, Link):
+                pid = item.data["package"]
+                fid = item.id
+            else:
+                raise TypeError("%s: Unknown item instance" % self.cname)
+            self.currentItem = (pid, fid)
+            self.log.debug0("%s.saveViewItemStates:currentItem:   name:'%s'   pid:%d   fid:%s" % (self.cname, item.data["name"], pid, str(fid)))
+        else:
+            self.log.debug0("%s.saveViewItemStates:currentItem: <none>" % self.cname)
+        # selection
+        self.selectedItems = []
+        for si in smodel.selectedRows(0):
+            item = si.internalPointer()
+            if isinstance(item, Package):
+                pid = item.id
+                fid = None
+            elif isinstance(item, Link):
+                pid = item.data["package"]
+                fid = item.id
+            else:
+                raise TypeError("%s: Unknown item instance" % self.cname)
+            self.selectedItems.append([pid, fid])
+            self.log.debug0("%s.saveViewItemStates:selectedItem:   name:'%s'   pid:%d   fid:%s" % (self.cname, item.data["name"], pid, str(fid)))
+        self.log.debug8("%s.saveViewItemStates took %dms" % (self.cname, self.time_msec() - func_start_time))
+    
+    def applyViewItemStates(self):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [CollectorModel.applyViewItemStates is too complex (24)](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)